ChartIQ/finsemble-seed
# Fixing build to support angular components

Angular components don't currently build themselves properly. This branch includes tweaks to the Gulp file to avoid copying angular component build envs to dist (can take a few mins) and then manually trigger the Angular component builds and output to dist instead.

Took an age to get right, but is essentially a pretty small set of changes. Speeds up the build a lot vs. having an angular component accidentally copied in full by gulp. Also ensures that components generated by CLI work, they did not previously due to issues with the output path (was going into /dist/components/myComponent/dist).

To test:

1. Add the angular CLI globally - DO NOT install it locally or Angular will refuse to generate the source for a nested component project (`You cannot use the new command inside an Angular CLI project.`):
    `npm install -g @angular/cli@latest`
2. Use the *finsemble* cli to generate an angular component:
    `finsemble-cli add component myAngularComponent --component-type=angular`
    Ignore the message:
    ```
    Unable to find "@angular/cli" in devDependencies.

    Please take the following steps to avoid issues:
    "npm install --save-dev @angular/cli@latest"
    ```
    As your generated component project will automatically have that dependency when generated.
3. Build and run with:
    `npm run dev`

The component generated will have a full angular build environment, including a large number of node_modules. It should auto-build itself when you hit `npm run dev` and install the output (no node_modules) in dist.
